contracts/ColonyPayment.sol

```diff
@@ -0,0 +1,119 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity >=0.5.3;
+pragma experimental "ABIEncoderV2";
+
+import "./ColonyStorage.sol";
+
+
+contract ColonyPayment is ColonyStorage {
+  function addPayment(address payable _recipient, address _token, uint256 _amount, uint256 _domainId, uint256 _skillId) public
+  stoppable
+  auth
+  domainExists(_domainId)
+  validPayoutAmount(_amount)
+  returns (uint256)
+  {
+    require(_recipient != address(0x0), "colony-payment-invalid-recipient");
+    paymentCount += 1;
+
+    fundingPotCount += 1;
+    fundingPots[fundingPotCount] = FundingPot({
+      associatedType: FundingPotAssociatedType.Payment,
+      associatedTypeId: paymentCount,
+      payoutsWeCannotMake: 1
```

Comment:
Only true if `_amount` is nonzero, right? Otherwise, I think that a payout that starts with a zero _amount can never finalize (because `payoutsWeCannotMake` will never be able to become zero).

Reply:
Very good catch, it slipped when I started allowing 0 amounts.. Not that is an excuse.